jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/assignment/impl/strategy/BusinessRuleAssignmentStrategy.java

```diff
@@ -0,0 +1,126 @@
+
+import java.util.Collection;
+import java.util.Date;
+import java.util.Set;
+import java.util.TreeSet;
+
+import org.drools.core.ClassObjectFilter;
+import org.kie.api.KieServices;
+import org.kie.api.builder.KieScanner;
+import org.kie.api.runtime.KieContainer;
+import org.kie.api.runtime.KieSession;
+import org.kie.api.runtime.rule.QueryResults;
+import org.kie.api.task.TaskContext;
+import org.kie.api.task.model.Task;
+import org.kie.internal.task.api.assignment.Assignment;
+import org.kie.internal.task.api.assignment.AssignmentStrategy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class BusinessRuleAssignmentStrategy implements AssignmentStrategy {
+
+    private static final Logger logger = LoggerFactory.getLogger(BusinessRuleAssignmentStrategy.class);    
+    private static final String IDENTIFIER = "BusinessRule";
+    
+    private boolean active = IDENTIFIER.equals(System.getProperty("org.jbpm.task.assignment.strategy"));
+
+    private String releaseId = System.getProperty("org.jbpm.task.assignment.rules.releaseId");
+    private String scannerInterval = System.getProperty("org.jbpm.task.assignment.rules.scan");
+    
+    
+    private KieServices kieServices = KieServices.Factory.get();
+    private KieContainer kieContainer;
+    private KieScanner kieScanner;
+        
+    public BusinessRuleAssignmentStrategy() {
+        
+        if (active) {                   
+            if (releaseId == null) {
+                throw new IllegalArgumentException("BusinessRule assignment strategy requires release id to be given via system property 'org.jbpm.task.assignment.rules.releaseId'");
+            }
+            String[] gav = releaseId.split(":");
+            logger.debug("Creating KieContainer for {} to be used for task assignments", releaseId);
+            this.kieContainer = kieServices.newKieContainer(kieServices.newReleaseId(gav[0], gav[1], gav[2]));
+            
+            if (scannerInterval != null) {
+                Long pollingInterval = Long.parseLong(scannerInterval);
+                logger.debug("Scanner to be enabled for {} container with polling interval set to {}", kieContainer, pollingInterval);
+                
+                this.kieScanner = this.kieServices.newKieScanner(kieContainer);
+                this.kieScanner.start(pollingInterval);
+                logger.debug("Scanner for container {} started at {}", kieContainer, new Date());
+            }
+        }
+    }
+    
+    @Override
+    public String getIdentifier() {
+        return IDENTIFIER;
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public Assignment apply(Task task, TaskContext context, String excludedUser) {
+        if (!active) {
+            logger.debug("{} strategy is not active", this);
+            return null;
+        }
+        logger.debug("Using rules to assign actual owner to task {}", task);
+        KieSession kieSession = this.kieContainer.newKieSession();
```

Comment:
I considered this at the beginning but that would require batch command to be used to isert, fire all reles etc so decided to go with default KieSession instead